pkg/trait/error_handler.go

```diff
@@ -67,6 +69,17 @@ func (t *errorHandlerTrait) Apply(e *Environment) error {
 			t.addErrorHandlerDependencies(e, defaultErrorHandlerURI)
 		}
 
+		if shouldHandleNoErrorHandler(e.Integration) {
+			// noErrorHandler is enabled by default on Kamelets since Camel 4.4.0 (runtimeVersion 3.8.0)
+			// need to disable this setting so that pipe error handler works
+			confValue, err := property.EncodePropertyFileEntry("camel.component.kamelet.noErrorHandler", "false")
+			if err != nil {
+				return err
+			}
+
+			e.Integration.Spec.AddConfigurationProperty(confValue)
```

Comment:
You need to add the property into the Environment and make sure this is called before the configmap holding `application.properties` is generated.

Reply:
sorry, can you please elaborate? I do not understand

Reply:
Something like `e.ApplicationProperties["key"]="value"`